presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeIcebergGeneralQueries.java

```diff
@@ -94,4 +104,11 @@ public void testDataSequenceNumberHiddenColumn()
                         .getOnlyValue(),
                 0L);
     }
+
+    @Test
+    public void testDateQueries()
+    {
+        assertQuery("SELECT * FROM ice_table_partitioned WHERE ds >= date ('1994-01-01')", "VALUES (1, date('2022-04-09')), (2, date('2022-03-18'))");
```

Comment:
All the queries (both insertion and reading) use the date function in predicates. The date function actually transforms the string date value to integer days since epoch. The constant folding will evaluate this function before creating the partition filter. So I feel this issue happens irrespective of Iceberg.

Can you try queries without date function in the filter predicate ? What happens ?

Reply:
That's how the date literals can be specified. I have removed the parenthesis to avoid the confusion. If we just specify the date string, then we get the varchar to date conversion error.